docs/design/offline-cascading-deletes/README.md

```diff
@@ -0,0 +1,414 @@
+| message          | message           | A JSON message.                                                                          | yes      |
+| timestamp        | timestamp         | Datetime stamp (Unix) when record created.                                               | yes      |
+| id               | id                | UUID v1, unique for the event (time-based is good here).                                 | yes      |
+| partitionKey     | partion_key       | UUID v4, typically the DocumentUuid of the document that caused generation of the event. | yes      |
+
+### Online Process - Outbox
+
+1. Receive PUT request.
+2. Check to see if the resource exists by looking up unique Document Uuid
+   (pulled from URL)
+    1. No: Return 404
+3. Yes - exists
+    1. Does calculated Meadowlark ID match the stored document?
+        1. Yes - proceed with normal update
+        2. No; Does the resource allow cascading?
+
+            :bulb: The MetaEd model defines which resources are allowed to have
+            natural key changes, with the language instruction `allow primary
+            key updates`. Thus it will be possible, in the API, to quickly
+            detect which resources are allowed to use the process described
+            above.
+
+            1. Yes - proceed to "START CASCADE UNIT OF WORK" below
+            2. No - update the document _except for the natural key_. Just
+               ignore the natural key update.
+    2. START CASCADE **UNIT OF WORK**:
+        1. Start atomic transaction
+        2. Update the document as requested, including the new Meadowlark ID.
+        3. Update all incoming references to hold the new Meadowlark ID, with
+           code something like:
+
+            ```none
+            # MongoDB
+            db.getCollection('documents').updateMany(
+              { 'outboundRefs': '<OLD VALUE>' },
+              { $set: { 'outboundRefs.$[elem]': '<NEW VALUE>' } },
+              {
+                upsert: false,
+                writeConcern: <CONFIG VALUE>,
+                arrayFilters: [ { 'elem': { $eq: '<OLD VALUE>' } } ]
+              }
+            )
+
+            # PostgreSQL
+            update references set parent_document_id = '<NEW VALUE>'
+              where parent_document_id = '<OLD VALUE>';
+            ```
+
+        4. Write a record to the new Outbox collection/table, like:
+
+              ```json
+              {
+                "eventType": "key-update",
+                "event": {
+                  "oldMeadowlarkId": <OLD VALUE>,
+                  "newMeadowlarkId": <NEW VALUE>,
+                  "newIdentity": {
+                    "sKeyPart1": "a",
+                    "sKeyPart2": "new-value",
+                  }
+                },
+                "timestamp": <current Unix timestamp>,
+                "id": <UUID v1>,
+                "partitionKey": <UUID of changed document>
+              }
```

Comment:
Note that this is only writing one record to the outbox. Because both the MongoDB and PostgreSQL update commands can operate on an entire set, the API application doesn't "know" which records are being modified. Therefore it can't write individual update records to the event stream.